core/modules/user.py

```diff
@@ -228,47 +228,46 @@ def login(self, name=None, password=None, skey="", *args, **kwargs):
 
         name = name.lower().strip()
         query = db.Query(self.userModule.viewSkel().kindName)
-        res = query.filter("name.idx >=", name).getEntry()
-
-        if res is None:
-            res = {"password": {"pwhash": "-invalid-", "salt": "-invalid-"}, "status": 0, "name": {}}
+        res = query.filter("name.idx >=", name).getEntry() or {}  # might find another user; always keep a dict
 
         password_data = res.get("password") or {}
         # old password hashes used 1001 iterations
         iterations = password_data.get("iterations", 1001)
-        passwd = encode_password(password, password_data.get("salt", ""), iterations)["pwhash"]
+        passwd = encode_password(password, password_data.get("salt", "-invalid-"), iterations)["pwhash"]
 
         # Check if the username matches
-        stored_user_name = (res.get("name") or {}).get("idx", "")
+        stored_user_name = (res.get("name") or {}).get("idx") or ""
         is_okay = secrets.compare_digest(stored_user_name, name)
 
         # Check if the password matches
-        stored_password_hash = password_data.get("pwhash", "-invalid-")
+        stored_password_hash = password_data.get("pwhash", b"-invalid-")
         is_okay &= secrets.compare_digest(stored_password_hash, passwd)
 
-        accountStatus: Optional[int] = None
+        status: Optional[int] = None
         # Verify that this account isn't blocked
-        if res["status"] < Status.ACTIVE.value:
+        if (res.get("status") or 0) < Status.ACTIVE.value:
```

Comment:
```suggestion
        if user_entry.get("status", 0) < Status.ACTIVE.value:
```
Here too and why not `user_entry.get("status", 0)`